### ISSUE TITLE
CheckBox Button - FormControl value is being updated twice when used with Reactive Forms

### ISSUE THREAD

**Issue by agugan** [primefaces/primeng#9608]
I'm submitting a 
```
[x] bug report
```
**Current behavior**

CheckBox form values are updated twice when used with reactive forms.

I think it is because along with registerOnChange, we have formControl.setValue being called explicitly when used with Reactive forms.
  

```
updateModel(event) {
....
            this.onModelChange(this.model);
            
            if (this.formControl) {
                this.formControl.setValue(this.model);
            }
}
 ....
  registerOnChange(fn: Function): void {
        this.onModelChange = fn;
    }
```

**Expected behavior**

CheckBox form values should be updated only once when used with reactive forms.

**Minimal reproduction of the problem with instructions**
https://stackblitz.com/edit/primeng-checkbox-bug

In the above stackblitz, if you check the console.log of `formControl.valueChanges` , you can see it is being logged twice.

- Angular version: 11.0.0

- PrimeNG version: 11.0.0-rc.1

- Browser: Tested in Chrome, Firefox

Happy to help with a PR.
Thanks for the awesome library.

**Comment by karol93** [comment]
same with template driven forms...

**Comment by jahanarun** [comment]
@yigitfindikli I also face the same issue. Any update on this would be appreciated.

**Comment by jahanarun** [comment]
@yigitfindikli Any updates on this?

**Comment by shybzzz** [comment]
the same issue but with AutoComplete
I would also much appreciate if this is fixed

**Comment by Ahmed-Hafez** [comment]
@mertsincan 
Hi Mertsincan,
I'm still facing the same issue.

**Comment by DarioVincenzoDeSimone** [comment]
I've found a potential fix for this bug by simply appending `pipe(debounceTime(500))` to the valueChanges observable (edit milliseconds as you wish)
While I acknowledge that this may not be a definitive solution, I hope it proves helpful.

**Comment by mehmetcetin01140** [comment]
Hi,

So sorry for the delayed response! Improvements have been made to many components recently, both in terms of performance and enhancement. Therefore, this improvement may have been developed in another issue ticket without realizing it. You can check this in the documentation. If there is no improvement on this, can you open a new issue so we can include it in our roadmap?

Thanks a lot for your understanding!
Best Regards,
